docs/DeveloperGuide.md

```diff
@@ -1,33 +1,297 @@
+* [`Commands`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/commands): The command executors.
+
+[`Exceptions`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/exceptions) represents exceptions used by multiple other components.
+
+### Overview
+
+The class diagram shows the relationship between `AthletiCLI`, `Ui`, `Parser`, and `Data`.
+
+<p  align="center" width="100%">
+  <img width="80%" src="images/MainClassDiagram.svg" alt="'set-diet-goal' Sequence Diagram"/>
+</p>
+
+### Data Component
+
+The class diagram shows how the `Data` component is constructed with multiple classes.
+
+<p  align="center" width="100%">
+  <img width="80%" src="images/DataClassDiagram.svg" alt="'set-diet-goal' Sequence Diagram"/>
+</p>
+
+**How the architecture components interact with each other**
+
+The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `help add-diet`.
+
+![](images/HelpAddDiet.svg)
+
+This diagram involves the interaction between `AthletiCLI`, `UI`, `Parser`, `Commands` components and the user.
+
+The `Storage` component only interacts with the `Data` component. The _Sequence Diagram_ below shows how they interact with each other for the scenario where a `save` command is executed.
+
+![](images/Save.svg)
+
+For simplicity, only 1 `StorableList` is drawn instead of the actual 6.
+
+## Implementation
+
+### Diet Management in AthletiCLI
+
+#### [Implemented] Setting Up, Editing, Deleting, Listing, and Finding Diets
+
+Regardless of the operation you are performing on diets (setting up, editing, deleting, listing, or finding), the process follows a general five-step pattern in AthletiCLI:
+
+1. **Input Processing**: The user's input is passed through AthletiCLI to the Parser Class. Examples of user inputs include:
+    - `add-diet calories/500 protein/20 carb/50 fat/10 datetime/2021-09-01 06:00` for adding a diet.
+    - `edit-diet 1 calories/500 protein/20 carb/50 fat/10 datetime/2021-09-01 06:00` for editing a diet.
+    - `delete-diet 1` for deleting a diet.
+    - `list-diet` for listing all diets.
+    - `find-diet 2021-09-01` for finding diets of a particular date.
+
+2. **Command Identification**: The Parser Class identifies the type of diet operation and passes the necessary parameters.
+
+3. **Command Creation**: An instance of the corresponding command class is created (e.g., AddDietCommand, EditDietCommand, etc.) and returned to AthletiCLI.
+
+4. **Command Execution**: AthletiCLI executes the command, interacting with the data instance of DietList to perform the required operation.
+
+5. **Result Display**: A message is returned post-execution and passed through AthletiCLI to the UI for display to the user.
+
+By following these general steps, AthletiCLI ensures a streamlined process for managing diet-related tasks.
+
+#### [Implemented] Setting Up of Diet Goals
```

Comment:
This diagram looks generally comprehensive but I think you the activation bar should end only if the self invocation is done. 
![IMG_1483](https://github.com/nus-cs2113-AY2324S1/tp/assets/139958049/1041a2de-8c96-443c-8030-2651ca20d752)

Reply:
Also I think it might be better that the cross signal indicating the end of the activation bar should be attached directly to the activation bar.